File: itmo/2018/10.11/f/f.py
v1, v2 = map(int, input().split())
l1, l2 = map(int, input().split())

a = l1 * v1 + l2 * v1
b1 = l1 * v1
c = v2 * (l1 + l2)
b2 = -l1 * v2

ans = 0
gg = 9e5
while ans < gg:
	if (ans * a + b1) % c == 0:
		print("YES")
		exit(0)
	if (ans * a + b2) % c == 0:
		print("YES")
		exit(0)
	ans += 1
print('NO')

File: sirius/march/day21/F.cpp
#include <iostream>
#include <vector>
#include <map>
#include <set>
#include <queue>
#include <algorithm>
#include <string>
#include <cmath>
#include <cstdio>
#include <iomanip>
#include <fstream>
#include <cassert>
#include <cstring>
#include <unordered_set>
#include <unordered_map>
#include <numeric>
#include <ctime>
#include <bitset>
#include <complex>

using namespace std;

int n;
vector<int> arr;
bool cmp(int a, int b){
    cout << "1 " << a << ' ' << b << '\n';
    string ans;
    cin >> ans;
    return ans == "YES";
}
inline void solve(){
    arr.resize(n);
    iota(arr.begin(), arr.end(), 1);
    stable_sort(arr.begin(), arr.end(), cmp);
    cout << "0 ";
    for (int i : arr)
        cout << i << ' ';
    cout << endl;
}


int main(){
    cin >> n;
    solve();
}
